Travis failing: Unable to locate perl command

Several recent PRs (including #6824, #6828, and #6829) are failing with the following error message:
```
=================================== FAILURES ===================================
_________________ TestFlagHandlers.test_no_build_system_flags __________________
self = <spack.test.flag_handlers.TestFlagHandlers object at 0x3dfaad0>
temp_env = None
    def test_no_build_system_flags(self, temp_env):
        # Test that both autotools and cmake work getting no build_system flags
        s1 = spack.spec.Spec('callpath')
        s1.concretize()
        pkg1 = spack.repo.get(s1)
>       spack.build_environment.setup_package(pkg1, False)
lib/spack/spack/test/flag_handlers.py:53: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
lib/spack/spack/build_environment.py:598: in setup_package
    dpkg.setup_dependent_package(pkg.module, spec)
var/spack/repos/builtin/packages/perl/package.py:186: in setup_dependent_package
    module.perl = self.spec['perl'].command
lib/spack/spack/spec.py:824: in __get__
    value = f()
lib/spack/spack/spec.py:817: in <lambda>
    callbacks_chain.append(lambda: self.default(self, instance, cls))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
descriptor = <spack.spec.ForwardQueryToPackage object at 0x142f2d0>
spec = perl@5.24.1%gcc@4.5.0+cpanm+shared arch=test-debian6-x86_64 ^gdbm@1.13%gcc@4.5..._glib arch=test-debian6-x86_64 ^readline@7.0%gcc@4.5.0 arch=test-debian6-x86_64
cls = <class 'llnl.util.lang.Spec'>
    def _command_default_handler(descriptor, spec, cls):
        """Default handler when looking for the 'command' attribute.
    
        Tries to search for ``spec.name`` in the ``spec.prefix.bin`` directory.
    
        Parameters:
            descriptor (ForwardQueryToPackage): descriptor that triggered the call
            spec (Spec): spec that is being queried
            cls (type(spec)): type of spec, to match the signature of the
                descriptor ``__get__`` method
    
        Returns:
            Executable: An executable of the command
    
        Raises:
            RuntimeError: If the command is not found
        """
        path = os.path.join(spec.prefix.bin, spec.name)
    
        if is_exe(path):
            return Executable(path)
        else:
            msg = 'Unable to locate {0} command in {1}'
>           raise RuntimeError(msg.format(spec.name, spec.prefix.bin))
E           RuntimeError: Unable to locate perl command in /home/travis/build/spack/spack/opt/spack/test-debian6-x86_64/gcc-4.5.0/perl-5.24.1-vy4kackxsavhhr7v3vgi5emsmyygu6cw/bin
```
This appears to be intermittent, especially in the case of #6824 where only a single version of Python exhibited this error.

Pinging @alalazo to see if he has any ideas.

--- a/lib/spack/spack/test/flag_handlers.py
+++ b/lib/spack/spack/test/flag_handlers.py
@@ -44,16 +44,16 @@ def add_o3_to_build_system_cflags(pkg, name, flags):
     return (flags, None, build_system_flags)
 
 
-@pytest.mark.usefixtures('config')
+@pytest.mark.usefixtures('config', 'mock_packages')
 class TestFlagHandlers(object):
     def test_no_build_system_flags(self, temp_env):
         # Test that both autotools and cmake work getting no build_system flags
-        s1 = spack.spec.Spec('callpath')
+        s1 = spack.spec.Spec('cmake-client')
         s1.concretize()
         pkg1 = spack.repo.get(s1)
         spack.build_environment.setup_package(pkg1, False)
 
-        s2 = spack.spec.Spec('libelf')
+        s2 = spack.spec.Spec('patchelf')
         s2.concretize()
         pkg2 = spack.repo.get(s2)
         spack.build_environment.setup_package(pkg2, False)
@@ -95,7 +95,7 @@ class TestFlagHandlers(object):
         assert 'SPACK_CPPFLAGS' not in os.environ
 
     def test_build_system_flags_cmake(self, temp_env):
-        s = spack.spec.Spec('callpath cppflags=-g')
+        s = spack.spec.Spec('cmake-client cppflags=-g')
         s.concretize()
         pkg = spack.repo.get(s)
         pkg.flag_handler = pkg.build_system_flags
@@ -109,7 +109,7 @@ class TestFlagHandlers(object):
         assert set(pkg.cmake_flag_args) == expected
 
     def test_build_system_flags_autotools(self, temp_env):
-        s = spack.spec.Spec('libelf cppflags=-g')
+        s = spack.spec.Spec('patchelf cppflags=-g')
         s.concretize()
         pkg = spack.repo.get(s)
         pkg.flag_handler = pkg.build_system_flags
@@ -134,7 +134,7 @@ class TestFlagHandlers(object):
             assert True
 
     def test_add_build_system_flags_autotools(self, temp_env):
-        s = spack.spec.Spec('libelf cppflags=-g')
+        s = spack.spec.Spec('patchelf cppflags=-g')
         s.concretize()
         pkg = spack.repo.get(s)
         pkg.flag_handler = add_o3_to_build_system_cflags
@@ -146,7 +146,7 @@ class TestFlagHandlers(object):
         assert pkg.configure_flag_args == ['CFLAGS=-O3']
 
     def test_add_build_system_flags_cmake(self, temp_env):
-        s = spack.spec.Spec('callpath cppflags=-g')
+        s = spack.spec.Spec('cmake-client cppflags=-g')
         s.concretize()
         pkg = spack.repo.get(s)
         pkg.flag_handler = add_o3_to_build_system_cflags
@@ -158,7 +158,7 @@ class TestFlagHandlers(object):
         assert pkg.cmake_flag_args == ['-DCMAKE_C_FLAGS=-O3']
 
     def test_ld_flags_cmake(self, temp_env):
-        s = spack.spec.Spec('callpath ldflags=-mthreads')
+        s = spack.spec.Spec('cmake-client ldflags=-mthreads')
         s.concretize()
         pkg = spack.repo.get(s)
         pkg.flag_handler = pkg.build_system_flags
@@ -174,7 +174,7 @@ class TestFlagHandlers(object):
         assert set(pkg.cmake_flag_args) == expected
 
     def test_ld_libs_cmake(self, temp_env):
-        s = spack.spec.Spec('callpath ldlibs=-lfoo')
+        s = spack.spec.Spec('cmake-client ldlibs=-lfoo')
         s.concretize()
         pkg = spack.repo.get(s)
         pkg.flag_handler = pkg.build_system_flags

